jax/_src/numpy/lax_numpy.py

```diff
@@ -4232,10 +4232,26 @@ def _eliminate_deprecated_list_indexing(idx):
   # deprecated by NumPy and exists for backward compatibility.
   if not isinstance(idx, tuple):
     if isinstance(idx, Sequence) and not isinstance(idx, ndarray):
+      # As of numpy 1.16, some non-tuple sequences of indices result in a warning, while
+      # others are converted to arrays, based on a set of somewhat convoluted heuristics
+      # (See https://github.com/numpy/numpy/blob/v1.19.2/numpy/core/src/multiarray/mapping.c#L179-L343)
+      # In JAX, we raise a warning for *all* non-tuple sequences, and in the future will
+      # *always* raise a TypeError here, rather than silently converting to an array or tuple
+      # depending on the contents of the list as numpy will. "Explicit is better than implicit".
+      # TODO(jakevdp): raise a TypeError here.
       if _any(_should_unpack_list_index(i) for i in idx):
+        msg = ("Using a non-tuple sequence for multidimensional indexing is deprecated; "
```

Comment:
Would it make sense to link to any NumPy deprecation plans?

Reply:
Initially I'd thought that, but given JAX's relative flexibility compared to numpy, I think it would be beneficial to deprecate list-based indices before numpy does. Then there will never be a time when the latest JAX and the latest numpy return different results for the same indexing operation.

Reply:
I mean not so much the concrete timeline, but more something that lays out the rationale for why it is to be deprecated (either our doc or theirs.)

Reply:
Done.